soh/soh/Enhancements/mods.cpp

```diff
@@ -600,6 +602,40 @@ void RegisterMirrorModeHandler() {
     });
 }
 
+f32 triforcePieceScale;
+
+void RegisterTriforceHunt() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        if (!GameInteractor::IsGameplayPaused()) {
+
+            // Warp to credits
+            if (GameInteractor::State::TriforceHuntCreditsWarpActive) {
+                gPlayState->nextEntranceIndex = 0x6B;
+                gSaveContext.nextCutsceneIndex = 0xFFF2;
+                gPlayState->sceneLoadFlag = 0x14;
+                gPlayState->fadeTransition = 3;
+                GameInteractor::State::TriforceHuntCreditsWarpActive = 0;
+            }
+
+            // Reset Triforce Piece scale for GI animation
+            if (GameInteractor::State::TriforceHuntPieceGiven) {
+                triforcePieceScale = 0.0f;
+                GameInteractor::State::TriforceHuntPieceGiven = 0;
+            }
```

Comment:
just curious as to why this is needed

Reply:
So I do some trickery to allow for the multiple triforce shard models. I'm syncing them by the count you've obtained, but because during the "get item" animation, 1 gets incremented shortly into the animation, I have to completely hide the triforce piece until it has time to increment and I can show the correct model (which matches the overworld model before you grab it).

So because I can't detect when a draw function just started being ran or not, I have to reset the scale and state somewhere else where I can detect when it's ran.

Reply:
if you could add some of that detail to the comment that'd be awesome!